components/org.wso2.carbon.database.utils/src/main/java/org/wso2/carbon/database/utils/jdbc/NamedTemplate.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.database.utils.jdbc;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.helpers.MessageFormatter;
+import org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class created to have Template which contains the methods for the database transactions is different from
+ * Template class as it supports the use of named prepared statement.
+ *
+ * @param <T> defines the input data type.
+ */
+public class NamedTemplate<T> {
```

Comment:
May be this argument is wrong :). Just trying to understand. Why do need to have a NamedTemplate and a NamedJDBCTemplate ? Do we need both ? To me NamedJDBCTemplate is pointless if we can move withTransaction to NamedTemplate

Reply:
There are two classes as Template and JdbcTemplate existing already. That's why I added named prepared statement support for both classes